src/dso_api/dynamic_api/serializers.py

```diff
@@ -207,32 +210,75 @@ def build_property_field(self, field_name, model_class):
     def to_representation(self, validated_data):
         data = super().to_representation(validated_data)
 
+        request = self.get_request()
+        from .urls import app_name
+
         # URL encoding of the data, i.e. spaces to %20, only if urlfield is present
         if self._url_content_fields:
             data = URLencodingURLfields().to_representation(
                 self._url_content_fields, data
             )
 
+        loose_relation_many_to_many_fields = [
+            f.attname
+            for f in self.Meta.model._meta.many_to_many
+            if isinstance(f, LooseRelationManyToManyField)
+        ]
+
+        for loose_relation_field_name in loose_relation_many_to_many_fields:
+            if not data[toCamelCase(loose_relation_field_name)]:
+                relation = self.Meta.model._meta.get_field(
+                    loose_relation_field_name
+                ).relation
+                dataset_name, table_name, relation_field_name = [
+                    to_snake_case(part) for part in relation.split(":")
+                ]
+                related_mgr = getattr(validated_data, loose_relation_field_name)
+                source_field_name = related_mgr.source_field_name
+                source_id = validated_data.id
+                through_tabel_filter_params = {source_field_name: source_id}
+                target_id_field = f"{related_mgr.target_field_name}_id"
+                through_tabel_items = [
+                    getattr(item, target_id_field, None)
+                    for item in related_mgr.through.objects.filter(
+                        **through_tabel_filter_params
+                    )
+                ]
+                result_list = [
+                    reverse(
+                        f"{app_name}:{dataset_name}-{table_name}-detail",
+                        kwargs={"pk": item},
+                        request=request,
+                    )
+                    for item in through_tabel_items
+                ]
+                data[toCamelCase(loose_relation_field_name)] = result_list
+
         if self.instance is not None:
-            if isinstance(self.instance, list):
-                # test workaround
-                model = self.instance[0]._meta.model
-            elif isinstance(self.instance, models.QuerySet):
-                # ListSerializer use
-                model = self.instance.model
-            else:
-                model = self.instance._meta.model
-            request = self.get_request()
-
-            for model_field in model._meta.get_fields():
-                permission_key = get_permission_key_for_field(model_field)
-                permission = request.auth_profile.get_read_permission(permission_key)
-                if permission is not None:
-                    key = toCamelCase(model_field.name)
-                    data[key] = mutate_value(permission, data[key])
+            data = self._profile_based_authorization_and_mutation(data)
```

Comment:
hier zou ik `data.update(self._get_profile_based_authorization())` oid doen. Het is niet nodig dit object heen en weer te sturen.

Reply:
Deze werd afgedwongen door Flake8 ;-). Goeie suggestie, de functie geeft nu een dict terug met alleen de gewijzigde values, waarmee vervolgens een update wordt gedaan.